src/searchModal.js

```diff
@@ -26,38 +26,59 @@ import component from 'ui/component';
 import 'ui/modal';
 import 'ui/datatable';
 import store from 'core/store';
-
+import resourceSelectorFactory from 'ui/resource/selector';
+import request from 'core/dataProvider/request';
+import urlUtil from 'util/url';
 /**
  * Creates a searchModal instance
  *
  * @param {object} config
  * @param {object} config.renderTo - DOM element where component will be rendered to
- * @param {string} config.query - search query to be set on component creation
+ * @param {string} config.criterias - Search criterias to be set on component creation
  * @param {boolean} config.searchOnInit - if init search must be triggered or not (stored results are used instead)
  * @param {string} config.url - search endpoint to be set on datatable
+ * @param {string} config.rootClassUri - Uri for the root class of current context, required to init the class filter
+ * @param {string} config.testMocks - mocks to be used for testing to avoid ajax requests
```

Comment:
oO What is this contraption? :o 

If you need to improve testability, please improve the API, or use mocks and stubs. Tweaking the API to support edge case testing is not a good practice, it may introduce weird behaviors. Moreover, testing using a featured hack will not certify the feature is working, at best it will demo your hack is working but no more.

If you need to mock the ajax request, please use mockjax ([example here](https://github.com/oat-sa/tao-core-ui-fe/blob/dab6658aad77dca8b7a7b5c21b07dbf8760a4318/test/datatable/test.js#L25)).

Reply:
Thanks, I had quite doubts about the best way to approach this. I knew about [mockserver](https://github.com/mock-server/mockserver), which [is used](https://github.com/oat-sa/test-taker-portal/blob/master/package.json#L50) on test takers portal,  but as I did not see in on `tao-core-ui-fe` I assumed it was not there for a reason and that I should not add it on my own. So I thought that without the possibility of intercepting AJAX request the only remaining possibility is what I tried.